app/src/main/java/com/sebastianvm/mango/ui/example/ExampleScreen.kt

```diff
@@ -1,38 +1,60 @@
 package com.sebastianvm.mango.ui.example
 
 import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
 import androidx.compose.material3.Button
+import androidx.compose.material3.ListItem
 import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
 
 @Composable
 fun ExampleScreen(exampleViewModel: ExampleViewModel, modifier: Modifier = Modifier) {
-    val viewState by exampleViewModel.state.collectAsStateWithLifecycle()
+    val viewState by exampleViewModel.stateFlow.collectAsStateWithLifecycle()
     ExampleScreenLayout(
         state = viewState,
-        onButtonClick = { exampleViewModel.handle(DidTapItem(it)) },
+        onSubmitJobName = { exampleViewModel.handle(JobNameEntered(it)) },
         modifier = modifier
     )
 }
 
 @Composable
 fun ExampleScreenLayout(
     state: ExampleState,
-    onButtonClick: (text: String) -> Unit,
+    onSubmitJobName: (text: String) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    Column(modifier = modifier) {
-        Text(text = state.displayText)
+    var jobName by remember {
```

Comment:
read some docs on `by remember`, If I'm understanding correctly does this cause a rerender of the example screen?

Reply:
Exactly, which is necessary for the text in the text field to be rerendered